src/main/java/com/example/event_manager/service/EventServiceImpl.java

```diff
@@ -40,13 +53,37 @@ public Event findById(final Long id) {
   }
 
   @Override
-  public Map<String, List<TaskStatus>> preapreTasksForEvent(final Event event) {
-    final Map<String, List<TaskStatus>> tasks = new HashMap<>();
-    final Set<TaskStatus> allTaskStatus = event.getTaskStatuses();
-    for (final TaskStatus ts : allTaskStatus) {
+  public EventForm eventFormById(final Long id) {
+    return eventMapper.fromEntity(findById(id));
+  }
+
+  @Override
+  public Map<String, List<TaskStatusForm>> preapreTasksForEvent(final EventForm event) {
+    final Map<String, List<TaskStatusForm>> tasks = new HashMap<>();
+    final List<TaskStatusForm> allTaskStatus = event.getTaskStatuses();
+    for (final TaskStatusForm ts : allTaskStatus) {
       tasks.computeIfAbsent(ts.getTaskStatusType(), k -> new ArrayList<>()).add(ts);
     }
 
     return tasks;
   }
+
+  @Override
+  public boolean deleteTaskStatusFromEvent(final Long taskId, final Long eventId) {
+    final Event event = findById(eventId);
+    final TaskStatus toDelete = taskStatusService.findById(taskId);
+    event.getTaskStatuses().remove(toDelete);
+    return save(event);
+  }
+
+  @Override
+  public void saveAdhocTaskToEvent(final TaskStatusForm taskStatusForm, final Long eventId) {
+    taskStatusForm.setTaskStatusType("adhoc");
```

Comment:
why taskStatusType is not enum?

Reply:
Jeżeli stworze nową listę predefiniowaną(o nazwie np. "siedziba sh") to do encji TaskStatusType (enum) musiałbym dodać nowy wiersz "siedziba sh".
 Po tworzeniu TaskStatus kolejny select po wszystkie enumy żebym do nowo stworzonych tasków mógł dodać ten typ. Nie mogę mieć
w kodzie na sztywno zdefiniowanego enuma bo kolejna list predefiniowana to kolejny wpis do enuma.
Prawdopodobnie myślałeś, że mamy rozróżnienie: adhoc, predef. U nas typem tasku jest albo adhoc albo nazwa listy predefiniowanej 
z ktorej task podchodzi. 
Wydaje mi się to trochę nienaturalne wiec czy dalej w to brnąć?